packages/orchestration/src/proposals/start-stakeAtom.js

```diff
@@ -16,49 +20,51 @@ const trace = makeTracer('StartStakeAtom', true);
  *     };
  *   };
  * }} powers
- * @param {{ options: StakeIcaTerms }} options
  */
-export const startStakeAtom = async (
-  {
-    consume: {
-      agoricNames,
-      board,
-      chainStorage,
-      chainTimerService,
-      orchestration,
-      startUpgradable,
-    },
-    installation: {
-      consume: { stakeIca },
-    },
-    instance: {
-      produce: { stakeAtom: produceInstance },
-    },
+export const startStakeAtom = async ({
+  consume: {
+    agoricNames,
+    board,
+    chainStorage,
+    chainTimerService,
+    orchestration,
+    startUpgradable,
   },
-  { options: { hostConnectionId, controllerConnectionId, bondDenom } },
-) => {
+  installation: {
+    consume: { stakeIca },
+  },
+  instance: {
+    produce: { stakeAtom: produceInstance },
+  },
+}) => {
   const VSTORAGE_PATH = 'stakeAtom';
-  trace('startStakeAtom', {
-    hostConnectionId,
-    controllerConnectionId,
-    bondDenom,
-  });
+  trace('startStakeAtom');
   await null;
 
   const storageNode = await makeStorageNodeChild(chainStorage, VSTORAGE_PATH);
   const marshaller = await E(board).getPublishingMarshaller();
   const atomIssuer = await E(agoricNames).lookup('issuer', 'ATOM');
   trace('ATOM Issuer', atomIssuer);
 
+  const chainHub = makeChainHub(await agoricNames);
+
+  const cosmoshub = await chainHub.getChainInfo('cosmoshub');
+  const connectionInfo = await chainHub.getConnectionInfo(
+    'agoriclocal',
+    'cosmoshub-4',
```

Comment:
I think these need to come in via config (the arg after powers) so that this can work in a starship config as well as mainnet, instagoric, etc.

Threading this sort of thing all the way thru the script and the getManifest call and such is tedious, but it seems necessary here.

Reply:
agree but I'm going to punt on Instagoric requirements